Test/VsVimTest/ReferenceTest.cs

```diff
@@ -192,7 +199,19 @@ internal static ReferenceData GetReferenceData(AssemblyName assemblyName)
                     case "Microsoft.VisualStudio.Platform.WindowManagement":
                         return new ReferenceData(assemblyName, ReferenceKind.ShellNonVersioned, getVersionFromVersionNumber());
 
+                    case MicrosoftCodeAnalysis:
+                    case MicrosoftCodeAnalysisCSharp:
+                    case MicrosoftCodeAnalysisCSharpScripting:
+                    case MicrosoftCodeAnalysisScripting:
+                    case SystemCollectionsImmutable:
+
+                        return new ReferenceData(assemblyName, ReferenceKind.ComponentVersioned, vsVersion: null);

```

Comment:
Don't think we should be taking this approach. Rather we should be mapping the versions of these DLLs to the VS version that they ship in. For instance for MS.CA 2.* ships in 2017, 3.* ships in 2019, etc ... That will make the rest of the validation remain unchanged.

Reply:
I made a correction, but I am not confident.
When I return from my company, I would like to continue working.

Reply:
I noticed that the comment of `ComponentVersioned` is not appropriate.

Reply:
That's what I was looking for here. Think this is good now. 

---
In reply to: [260988572](https://github.com/jaredpar/VsVim/pull/2371#discussion_r260988572) [](ancestors = 260988572)

Reply:
Yeah we should update that comment but overall this is good now. 

---
In reply to: [260989748](https://github.com/jaredpar/VsVim/pull/2371#discussion_r260989748) [](ancestors = 260989748)